src/fluree/db/dbfunctions/ctx.cljc

```diff
@@ -0,0 +1,85 @@
+(ns fluree.db.dbfunctions.ctx
+  (:require [fluree.db.util.json :as json]
+            [fluree.db.dbfunctions.core :as dbfunctions]
+            [fluree.db.dbfunctions.fns :refer [extract]]
+            [fluree.db.util.async :refer [<? go-try channel?]]
+            [fluree.db.permissions-validate :as perm-validate]
+            [fluree.db.util.core :as util]
+            [fluree.db.query.range :as query-range]
+            [fluree.db.util.log :as log]
+            [fluree.db.constants :as const]
+            #?(:cljs [fluree.db.flake :refer [Flake]])
+            [fluree.db.dbproto :as dbproto])
+  #?(:clj (:import (fluree.db.flake Flake))))
+
+;; Handles context
+
+(defn ctx-flakes->k+fn
+  "Iterates over ctx-flakes to extract context key and context fn subject id as two-tuple"
+  [ctx-flakes]
+  (when (seq ctx-flakes)
+    (loop [[^Flake f & r] ctx-flakes
+           k nil
+           v nil]
+      (if (and k v)
+        [k v]
+        (when f
+          (cond
+            (= const/$_ctx:key (.-p f))
+            (recur r (.-o f) v)
+
+            (= const/$_ctx:fn (.-p f))
+            (recur r k (.-o f))
+
+            :else
+            (recur r k v)))))))
+
+
+(defn resolve-ctx
+  "Resolves value for a single ctx for a given user."
+  [db-root ?ctx ctx-sid]
+  (go-try
+    (let [ctx-flakes (<? (query-range/index-range db-root :spot = [ctx-sid]))
+          [k fn-sid] (ctx-flakes->k+fn ctx-flakes)
+          ctx-fn-str (some-> (<? (query-range/index-range db-root :spot = [fn-sid const/$_fn:code]))
+                             first
+                             (#(.-o ^Flake %)))
```

Comment:
You could use these fns instead to clean up these threading macros and get rid of the type hints in here (and potentially the need to import the `Flake` class altogether depending on what the reflection warnings say): https://github.com/fluree/db/blob/38bbd69638dada151ac29d1d58a058551efe3bf2/src/fluree/db/flake.cljc#L173-L195